### PR TITLE
chore: RPC-related Cleanup

### PR DESCRIPTION
**Pull request by harrysolovay**

### CODE REVIEW

File: effect/std/atoms/rpcClient.ts [hunk 1]
```diff
@@ -2,14 +2,17 @@ import * as rpc from "../../../rpc/mod.ts";
 import { effector, EffectorArgs } from "../../impl/mod.ts";
 
 export interface RpcClientR<Beacon> {
-  rpc: rpc.RpcClientFactory<Beacon>;
+  rpc: rpc.RpcClientFactory<Beacon, rpc.RpcError>;
```
Comment (harrysolovay) [inline effect/std/atoms/rpcClient.ts:5]: Ignore the fact that the RPC errors aren't piped in through the top-level `rpcClient` effect. We'll get to this later (#46).

File: rpc/Base.ts [hunk 2]
```diff
@@ -8,41 +9,84 @@ export type ListenerCb<IngressMessage_ extends M.IngressMessage = M.IngressMessa
...
-    super();
-  }
-}
+export abstract class RpcClient<RpcError extends E.RpcError> {
+  #nextId = 0;
+  listeners = new Map<ListenerCb, boolean>();
+  #errors: RpcError[] = [];
 
-export abstract class RpcClient {
+  /** Send a message to the RPC server without
```
Comment (tjjfvi) [inline rpc/Base.ts:17]: without what?

File: rpc/Base.ts [hunk 3]
```diff
@@ -8,41 +9,84 @@ export type ListenerCb<IngressMessage_ extends M.IngressMessage = M.IngressMessa
 
 export type StopListening = () => void;
 
-export class RpcError extends Error {
-  constructor(readonly inner: M.ErrMessage) {
-    super();
-  }
-}
+export abstract class RpcClient<RpcError extends E.RpcError> {
```
Comment (tjjfvi) [inline rpc/Base.ts:12]: Why is this generic on the error?

Comment (harrysolovay) [reply rpc/Base.ts:12]: My thinking is that we'll want beacon-specific error types (web sockets can error out differently from Smoldot client instances).

**Pull request merged**